dc-commons-file/src/main/java/de/digitalcollections/commons/file/backend/impl/FileResourceRepositoryImpl.java

```diff
@@ -260,4 +279,79 @@ public void write(FileResource resource, String input) throws ResourceIOExceptio
     }
   }
 
+  @Override
+  public Set<String> findKeys(String keyPattern, FileResourcePersistenceType resourcePersistenceType) throws ResourceIOException {
+    // The pattern for valid keys is the original pattern without any brackets inside, but surrounded with one bracket.
+    Pattern validKeysPattern = Pattern.compile(
+        "("
+            + keyPattern.replace("(","").replace(")","").replace("^","").replace("$","")
```

Comment:
what if more open brackets, better use replaceAll?

Reply:
String.replace replaces all occurrences of the term; String.replaceAll does the same, but with aregular expression